### PR TITLE
Change default PiHole Server IP to 0.0.0.0

### PR DESCRIPTION
**Pull request by dylanmtaylor**
This is a possible solution for #1056. If the user does not set the server IP address they will currently get a confusing error and it will not allow docker-compose to run. This should at least get the container to build correctly. A message indicating that this should be set would also be greatly beneficial to users.

### CODE REVIEW

**Comment by dylanmtaylor** [comment]
Testing done:

* I set the PIHOLE_SERVER_IP variable to 0.0.0.0 and re-ran compose, which worked successfully
* Using the new configuration, I was able to navigate to localhost:8008/admin and access PiHole's web interface

I believe this is a better default than compose not working at all.

**Comment by dylanmtaylor** [comment]
@nemchik This one is ready for review as well

**Comment by nemchik** [comment]
This feels like a good change, but the reason we don't have it default to this already is it still can be a conflict, often with resolv listening on 53, but this could work out of the box for some. All of the documentation on our pihole page is out of date. The current setup (before this change) doesn't require any of the current instruction on the current pihole documentation page. Would you be willing to update that page and replace everything with instructions relevant to this change? We would want to include the idea that you can set the IP to your server IP address (ex `192.168.1.5` or whatever) in case `0.0.0.0` doesn't work out of the box. Basically there should be no need to modify the host system, just the docker variables.

**Comment by dylanmtaylor** [comment]
> Would you be willing to update that page and replace everything with instructions relevant to this change?

Sure, let me take a look at the docs and see what needs changed.

**Comment by dylanmtaylor** [comment]
@nemchik - I made a commit with new documentation, what do you think of that?

File: docs/apps/pihole.md [hunk 1]
```diff
@@ -5,52 +5,8 @@
...
-
-### Name resolution for localhost
-
-In most cases it might be required to set your localhost name in `/etc/hosts`
-
-```hosts
-127.0.0.1    machinename.localhost    machinename
-127.0.0.1    domain.com
-```
+By default, Pi-hole is configured to bind to `0.0.0.0` on port 53. This configuration should work out of the box for many people, however, if you encounter issues with this, you may need to disable or reconfigure any conflicting services running on port 53, such as `systemd-resolved` or set the `PIHOLE_SERVER_IP` variable to your server's IP address (e.g. 192.168.1.5). To stop systemd-resolved from listening on port 53, you can set `DNSStubListener=no` in `/etc/systemd/resolved.conf`.
```
Comment (nemchik) [inline docs/apps/pihole.md:12]: I love how minimal this is, but I would take it a step further and recommend the `PIHOLE_SERVER_IP` option first and then list the option to modify the resolved.conf as an alternative that may be needed for advanced configurations.

**Comment by nemchik** [comment]
By the way are you on our discord?

**Comment by dylanmtaylor** [comment]
> By the way are you on our discord?

Yes, as Dylan#9414

**Pull request merged**